src/Cake/Infrastructure/Scripting/ReferenceAssemblyResolver.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Cake.Core.Diagnostics;
+using Cake.Core.Scripting;
+
+namespace Cake.Infrastructure.Scripting
+{
+    public sealed class ReferenceAssemblyResolver : IReferenceAssemblyResolver
+    {
+        private readonly ICakeLog _log;
+
+        public ReferenceAssemblyResolver(ICakeLog log)
+        {
+            _log = log;
+        }
+
+        public Assembly[] GetReferenceAssemblies()
+        {
+            IEnumerable<Assembly> TryGetReferenceAssemblies()
+            {
+                foreach (var reference in
+#if NETCOREAPP3_1
+            Basic.Reference.Assemblies.NetCoreApp31.All)
+#elif NET5_0
+            Basic.Reference.Assemblies.Net50.All)
+#else
+            Basic.Reference.Assemblies.Net60.All)
+#endif
+                {
+                    Assembly assembly;
+                    try
+                    {
+                        assembly = Assembly.Load(System.IO.Path.GetFileNameWithoutExtension(reference.FilePath));
+                    }
+                    catch (Exception ex)
+                    {
+                        _log.Debug(log => log("Failed to load {0}\r\n{1}", reference.FilePath, ex));
```

Comment:
Should `\r\n` be replaced with something portable? Actually, I see `\r\n` is sprinkled throughout the Cake source, perhaps inconsistently?

Reply:
@gitfool This is definitely something that can be improved, but given that this is a debug log, I don't think this is a showstopper.  If you have found this being used in other places, would you be up for a maintenance PR to fix it up in all places?  For now, I will resolve this conversation.